03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+- Be wary of overplotting. If your plot has too many dots or lines and it starts to look like a mess, then you need to do something different.
+- Only make the plot area (where the dots, lines, bars are) as big as needed. Simple plots can be made small.
+- Don't adjust the axes to zoom in on small differences. If the difference is small, show that it's small!
+
+
+## Creating visualizations with `ggplot2` 
+#### *Build the visualization iteratively* {-#my-section}
+
+This section will cover examples of how to choose and refine a visualization given a dataset and a question that you want to answer, 
+and then how to create the visualization in R using `ggplot2`.
+
+### The Mauna Loa CO2 dataset 
+ This dataset contains the atmospheric concentration of carbon dioxide (CO2, in parts per million) at the Mauna Loa research station in Hawaii 
+from the years 1959-1997. **Question:** Does the concentration of atmospheric CO2 change over time, and are there any interesting patterns to note?
+```{r 03-data-co2}
+# mauna loa carbon dioxide data 
+# (with some light wrangling to make it a dataframe with informative column names)
+co2_df <- data.frame(concentration = as.matrix(co2), date = time(co2))
+head(co2_df)
+```
+
+Since we are investigating a relationship between two variables (CO2 concentration and date), a scatter plot is a good place to start. Scatter plots 
+show the data as individual points with `x` (horizonal axis) and `y` (vertical axis) coordinates. Here, we will use the date as the `x` coordinate 
+and CO2 concentration as the `y` coordinate. When using the `ggplot2` library, we create the plot object with the `ggplot` function; there are 
+a few basic aspects of a plot that we need to specify:
+
+- the *data*: the name of the dataframe object that we would like to visualize 
+    - here, we specify the `co2_df` dataframe
+- the *aesthetic mapping*: tells `ggplot` how the columns in the dataframe map to properties of the visualization
+    - to create an aesthetic mapping, we use the `aes` function
+    - here, we set the plot `x` axis to the `date` variable, and the plot `y` axis to the `concentration` variable
+- the *geometric object*: specifies how the mapped data should be displayed
+    - to create a geometric object, we use a `geom_*` function (see the [ggplot reference](https://ggplot2.tidyverse.org/reference/) for a list of geometric objects)
+    - here, we use the `geom_point` function to visualize our data as a scatterplot
+
+There are many other possible arguments we could pass to the aesthetic mapping and geometric object to change how the plot looks. For 
+the purposes of quickly testing things out to see what they look like, though, we can just go with the default settings:
+
+```{r 03-data-co2-scatter}
+co2_scatter <- ggplot(co2_df, aes(x = date, y = concentration)) + geom_point() 
+```
+```{r 03-data-co2-scatter-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
```

Comment:
Don't hide the print call. I think it's important to explain that if you save a plot to an object (and you should, and we should also explain why) you need to call the plot to have it show up. In Jupyter & R Markdown just the plot name is sufficient (print is not needed).

Reply:
done